Mathlib/Analysis/Polynomial/MahlerMeasure.lean

```diff
@@ -137,8 +139,117 @@ theorem prod_mahlerMeasure_eq_mahlerMeasure_prod (s : Multiset ℂ[X]) :
   | empty => simp
   | cons _ _ ih => simp [mahlerMeasure_mul, ih]
 
-theorem logMahlerMeasure_mul_eq_add_logMahelerMeasure {p q : ℂ[X]} (hpq : p * q ≠ 0) :
+theorem logMahlerMeasure_mul_eq_add_logMahlerMeasure {p q : ℂ[X]} (hpq : p * q ≠ 0) :
     (p * q).logMahlerMeasure = p.logMahlerMeasure + q.logMahlerMeasure := by
   simp_all [logMahlerMeasure_eq_log_MahlerMeasure, mahlerMeasure_mul, log_mul]
 
+theorem logMahlerMeasure_C_mul {a : ℂ} (ha : a ≠ 0) {p : ℂ[X]} (hp : p ≠ 0) :
+    (C a * p).logMahlerMeasure = log ‖a‖ + p.logMahlerMeasure := by
+  rw [logMahlerMeasure_mul_eq_add_logMahlerMeasure (by simp [ha, hp]), logMahlerMeasure_const]
+
+open MeromorphicOn Metric in
+/-- The logarithmic Mahler measure of `X - C z` is the `log⁺` of the absolute value of `z`. -/
+@[simp]
+theorem logMahlerMeasure_X_sub_C (z : ℂ) : (X - C z).logMahlerMeasure = log⁺ ‖z‖ := by
+  rcases eq_or_ne z 0 with rfl | hz₀
+  · simp
+  have hmeroOnB : MeromorphicOn (· - z) (closedBall 0 |1|) := id.sub <| const z
+  simp_rw [logMahlerMeasure_def, eval_sub, eval_X, eval_C]
+  rw [circleAverage_log_norm one_ne_zero hmeroOnB, abs_one,
+  -- get rid of the `meromorphicTrailingCoeffAt`
+    AnalyticAt.meromorphicTrailingCoeffAt_of_ne_zero (by fun_prop) (by simp [hz₀]),
+    ← finsum_mem_support]
+  set B := closedBall (0 : ℂ) 1
+  -- separate cases depending on whether z is in the open `ball 0 1` or not
+  by_cases hzBall : z ∈ ball 0 1
+  · have hzdiv1 := divisor_sub_const_self (show z ∈ B by grind [ball_subset_closedBall])
+    suffices (fun u ↦ -(divisor (· - z) B u * log ‖u‖)).support = {z} by
+      simp [this, hzdiv1, posLog_def]
+      grind [mem_ball_zero_iff, log_nonpos, norm_nonneg]
+    rw [Function.support_eq_iff]
+    constructor
+    · simp [hzdiv1]
```

Comment:
Non terminal `simp`. (it's probably fine, but let's see if it's reasonable to squeeze it)

Reply:
Yes, that's why I said it's probably fine. Do as you prefer.